go-controller/pkg/cluster/master.go

```diff
@@ -170,10 +170,20 @@ func parseNodeHostSubnet(node *kapi.Node) (*net.IPNet, error) {
 	return subnet, nil
 }
 
-func (cluster *OvnClusterController) ensureNodeLogicalNetwork(nodeName string, hostsubnet *net.IPNet) error {
+func (cluster *OvnClusterController) ensureNodeLogicalNetwork(node *kapi.Node, hostsubnet *net.IPNet) error {
 	ip := util.NextIP(hostsubnet.IP)
 	n, _ := hostsubnet.Mask.Size()
 	firstIP := fmt.Sprintf("%s/%d", ip.String(), n)
+	var clusterIPSubnet []string
+	nodeName := node.Name
+	macAddress := node.Annotations[OvnNodeMacAddress]
+	ifaceID := node.Annotations[OvnNodeIfaceID]
+	localnetGatewayIP := node.Annotations[OvnNodeLocalnetGatewayIP]
+	localnetGatewayNextHop := node.Annotations[OvnNodeLocalnetGatewayNextHop]
+	subnet := node.Annotations[OvnNodeSubnet]
+
+	err := util.GatewayInit(clusterIPSubnet, nodeName, ifaceID, localnetGatewayIP,
```

Comment:
I would do this bit in a separate function like "syncGatewayLogicalNetwork()" like @mccv1r0 did in https://github.com/ovn-org/ovn-kubernetes/pull/767 . And we'll need validation of each annotation too (like for macAddress you want to net.ParseHardwareAddr() and return an error if fails; same for the InterfaceIP and InterfaceNextHop). And if any of them are missing, return.

Then we also call this function from UpdateFunc in watchNodes() in addition to AddFunc so that we ensure that if one annotation is missing whenever it gets added then syncGatewayLogicalNetwork() would get called again and complete successfully.